File: solutions/lua/hamming/1/hamming.lua
local Hamming = {}

function Hamming.compute(a,b)
    if string.len(a)~=string.len(b)
    then
        return -1
    end
    v=0
    for i=1,string.len(a),1
    do
        if string.sub(a,i,i)~=string.sub(b,i,i)
        then
            v=v+1
        end
    end
    return v        
end

return Hamming

File: solutions/javascript/hamming/1/hamming.js
//
// This is only a SKELETON file for the 'Hamming' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const compute = (a, b) => {
  var ret = 0;
  if (a.length != b.length)
    throw new Error('strands must be of equal length');
  for (var i = 0; i<a.length; i++){
    if (a.charAt(i) != b.charAt(i)) ret++
  }
  return ret;
  //throw new Error('Remove this statement and implement this function');
};
